tests/test_core.py

```diff
@@ -4558,7 +4558,13 @@ def test_fs_errorstack(self):
       #include <emscripten.h>
       int main(void) {
         EM_ASM(
-          FS.write('/dummy.txt', 'homu');
+          "use strict";
```

Comment:
Oh interesting, so the issue was that binaryen.js runs in strict mode?

For more coverage, how about adding the `use strict` at the very top level? You can use a `post_build` in the test runner to add it. 

Also, I see I can recreate the binaryen.js error by building `tests/hello_libcxx.cpp` and adding `use strict` (since iostream code ends up using enough filesystem things to trigger the error), so testing that way would be good I think, it would ensure strict mode works on a complete nontrivial program.

Reply:
>For more coverage, how about adding the use strict at the very top level? You can use a `post_build` in the test runner to add it.

I think we should do it on resulting script by default, not only when testing.

Reply:
It's tricky for another reason too, strict mode has been slower than normal mode. I'm not sure if that's still the case or not, but it used to be a big difference.

Reply:
Someone created [a short test case](https://jsperf.com/strict-mode) and it seems strict mode is faster at least for some cases on Chrome.

![image](https://user-images.githubusercontent.com/3396686/34460322-0fc6a2d4-ee4e-11e7-9de9-63a626b86067.png)

Although the strict mode is in fact slightly slower (with differences < 10%) on Firefox and MSEdge.

Reply:
Interesting, thanks. I still worry though that perf can be very variable with various features. But in any case, the bigger issue is the one you mentioned before, about pre-js and js library code from users potentially breaking due to this, that seems too risky.